python/pyarrow/filesystem.py

```diff
@@ -431,7 +447,18 @@ def _ensure_filesystem(fs):
             # In case its a simple LocalFileSystem (e.g. dask) use native arrow
             # FS
             elif mro.__name__ == 'LocalFileSystem':
-                return LocalFileSystem.get_instance()
+                return LocalFileSystem._get_instance()
+
+        try:
+            import fsspec
```

Comment:
`fsspec` seems to take some time to import, we should probably lookup directly in `sys.modules` instead.